src/Authentication/Authentication/Cmdlets/InvokeGraphRequest.cs

```diff
@@ -118,65 +262,704 @@ protected override void ProcessRecord()
+                        var fileName = response.Content.Headers.ContentDisposition.FileNameStar;
+                        var fullFileName = QualifyFilePath(fileName);
+                        WriteVerbose(string.Format(CultureInfo.InvariantCulture, "Inferred File Name {0} Saving to {1}", fileName, fullFileName));
+                        StreamHelper.SaveStreamToFile(baseResponseStream, fullFileName, this,
+                            _cancelToken.Token);
+                    }
+                }
+                else
+                {
+                    WriteVerbose("Could not Infer File Name");
+                }
+            }
+
+            if (!string.IsNullOrEmpty(StatusCodeVariable))
+            {
+                var vi = SessionState.PSVariable;
+                vi.Set(StatusCodeVariable, (int)response.StatusCode);
+            }
+
+            if (!string.IsNullOrEmpty(ResponseHeadersVariable))
+            {
+                var vi = SessionState.PSVariable;
+                vi.Set(ResponseHeadersVariable, WebResponseHelper.GetHeadersDictionary(response));
+            }
+        }
+
+        private static bool TryConvert<T>(string str, out object obj, ref Exception exRef)
+        {
+            var converted = false;
+            try
+            {
+                obj = JsonConvert.DeserializeObject<T>(str);
+                if (obj == null)
+                    JToken.Parse(str);
+                else
+                    converted = true;
+            }
+            catch (JsonException ex)
+            {
+                var msg = string.Format(CultureInfo.CurrentCulture, "JsonDeserializationFailed", ex.Message);
+                exRef = new ArgumentException(msg, ex);
+                obj = null;
+            }
+            catch (Exception jsonParseException)
+            {
+                exRef = jsonParseException;
+                obj = null;
+            }
+
+            return converted;
+        }
+
+        private IAuthenticationProvider GetAuthProvider()
+        {
+            if (Authentication == GraphRequestAuthenticationType.UserProvidedToken)
+            {
+                return new InvokeGraphRequestAuthProvider(GraphRequestSession);
+            }
+
+            return AuthenticationHelpers.GetAuthProvider(GraphSession.Instance.AuthContext);
+        }
+
+        private HttpClient GetHttpClient()
+        {
+            var provider = GetAuthProvider();
+            var client = HttpHelpers.GetGraphHttpClient(provider);
+            return client;
+        }
+
+        private HttpResponseMessage GetResponse(HttpClient client, HttpRequestMessage request)
+        {
+            if (client == null) throw new ArgumentNullException(nameof(client));
+
+            if (request == null) throw new ArgumentNullException(nameof(request));
+
+            var cancellationToken = _cancelToken.Token;
+            var response = client.SendAsync(request, HttpCompletionOption.ResponseHeadersRead, cancellationToken)
```

Comment:
We don't need the cancellation token, it isn't ever cancelled by the user or us. You could add some sort of function or script block parameter that would be cancel the operation on Invoke-GraphRequest. I think we can expect a user will just ctrl+break.  Use the overload without the cancellation token and remove references to the cancellation source.

Reply:
We should cancel the `CancellationTokenSource` in the `StopProcessing` override method. `StopProcessing` is the method that's called on `ctrl+break`. See the implementation here https://github.com/microsoftgraph/msgraph-sdk-powershell/blob/dev/src/Authentication/Authentication/Cmdlets/ConnectGraph.cs#L140